src/docs/release/breaking-changes/platform-views-using-html-slots-web.md

```diff
@@ -0,0 +1,196 @@
+elements are used to represent platform views. The actual content of each
+platform view is rendered *outside* of the Shadow Root, as a child of the Shadow
+Root host.
+
+### Before
+
+<!-- skip -->
+```html
+...
+
+<flt-glass-pane>
+  ...
+  <div id="platform-view">Contents</div> <!-- canvaskit -->
+  <!-- OR -->
+  <flt-platform-view>
+    #shadow-root
+    | <div id="platform-view">Contents</div> <!-- html -->
+  </flt-platform-view>
+  ...
+</flt-glass-pane>
+
+...
+```
+
+### After
+
+<!-- skip -->
+```html
+...
+
+<flt-glass-pane>
+  #shadow-root
+  | ...
+  | <flt-platform-view-slot>
+  |   <slot name="platform-view-1" />
+  | </flt-platform-view-slot>
+  | ...
+  <flt-platform-view slot="platform-view-1">
+    <div id="platform-view">Contents</div>
+  </flt-platform-view>
+  ...
+</flt-glass-pane>
+
+...
+```
+
+Now, when the framework needs to move DOM nodes around, it'll operate over
+`flt-platform-view-slot`s, which only contain the `slot` tag that "projects" the
+content defined in the `flt-platform-view` tags outside of the Shadow Root
+(which never move), thus preventing the reloads.
+
+From an app's perspective, this change is transparent, **however** this was
+deemed a _breaking change_ because tests that make assumptions about the DOM of
+a Flutter web app can break.
+
+## Migration guide
+
+### Code
+
+The content returned by the users' Platform View Factory used to be resized and
+positioned by the framework, but not anymore. Ensure that your returned
+`html.Element` takes the whole space allocated for it by the framework, by
+setting its `style.width` and `style.height` properties to `'100%'`:
+
+<!-- skip -->
+```dart
+ui.platformViewRegistry.registerViewFactory(viewType, (int viewId) {
+  final html.Element htmlElement = html.DivElement()
+    // ..other props
+    ..style.width = '100%'
+    ..style.height = '100%';
+  // ...
+  return htmlElement;
+});
+```
+
+Otherwise the engine will print a warning message to the console similar to:
+
+<!-- skip -->
+```bash
+Height of Platform View type: [$viewType] may not be set. Defaulting to `height: 100%`.
+Set `style.height` to any appropriate value to stop this message.
+```
+
+or:
+
+<!-- skip -->
+```bash
+Width of Platform View type: [$viewType] may not be set. Defaulting to `width: 100%`.
+Set `style.width` to any appropriate value to stop this message.
+```
+
+### Tests
+
+Unfortunately, there are no selectors that let one _pierce_ through the Shadow
+DOM boundary to locate elements underneath it, so code that once peeked inside
+of the `flt-glass-pane`, needs to be made Shadow DOM aware. Using the html
+markup from above:
+
+<!-- skip -->
+```dart
+// Find #some-element inside the app:
+final element = document.querySelector('#some-element');
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+final shadowRoot = document.querySelector('flt-glass-pane')?.shadowRoot!;
+final element = shadowRoot.querySelector('#some-element');
```

Comment:
Why is this necessary? After the migration `#some-element` will no longer be inside the shadow DOM, only the slot. We don't want people looking into Flutter's private DOM structure. It's unpredictable and unstable.

Reply:
This is what most of the tests that failed were doing. We can document the fact that now they shouldn't drill inside the shadowDOM, as mentioned above, that's probably better!

Reply:
Removed this, warned people against looking into the shadowDom of the app.